src/main/scala/io/iohk/ethereum/vm/VM.scala

```diff
@@ -123,6 +123,8 @@ class VM[W <: WorldStateProxy[W, S], S <: Storage[S]] extends Logger {
     } else if (codeStoreOutOfGas && !config.exceptionalFailedCodeDeposit) {
       // Code storage causes out-of-gas with exceptionalFailedCodeDeposit disabled
       result
+    } else if(result.error.contains(RevertOccurs)) {
```

Comment:
I am not sure this is good place for this condition. Main reason is that if `Revert` would occur, but the conditions in 
https://github.com/input-output-hk/mantis/pull/493/files#diff-5b7d8817413ce869c362b5398e9a65f1L120 would be fullfilled we would return `result.copy(error = Some(OutOfGas), gasRemaining = 0)` so that is definitly not our intention.
Example of revert handling in that place in geth: https://github.com/ethereum/go-ethereum/blob/1da33028ce88c4365d99471977098f4911fd38fa/core/vm/evm.go#L383

Reply:
case of do not have enough gas is coveret here : https://github.com/input-output-hk/mantis/blob/284578afeffaeefc0fd7368976d51008ff786bcf/src/main/scala/io/iohk/ethereum/vm/OpCode.scala#L196. 

In this place we have create already returned and i think w do not want to change it to `OutOfGas`  by accident (later we will need to extract return data from revert and pass it to return buffer)

Reply:
Still, we have to check if reversion occurs, because we don't want to deploy the code and don't want to spend gas for code deposit. I think `saveNewContract` is the best place to do that. I could move this logic one level up but it won't change anything.

Reply:
I agree that this method is good place to handle possible errors, but i want to avoid risk of changing the returned error from `Revert` to `outOfGas`. I mean if the executions of contract code ended with `Revert` it needs to be propagatet further even if `maxCodeSizeExceeded || (codeStoreOutOfGas && config.exceptionalFailedCodeDeposit` is true